internal/service/vault.go

```diff
@@ -51,3 +52,70 @@ func RetrieveDeck(id string, lang string) (*vault.DeckVault, error) {
 
 	return &vaultDeck, nil
 }
+
+func RetrieveRandomDeckId(set int) (string, error) {
+	lastDeck, err := retrieveLastDecks(set, 1, 0)
+	if err != nil {
+		log.Error(err.Error())
+		return "", err
+	}
+
+	deckOffset := rand.Intn(lastDeck.Counter-1) + 1
+	deck, err := retrieveLastDecks(set, 1, deckOffset)
+	if err != nil {
+		log.Error(err.Error())
+		return "", err
+	}
+
+	if len(deck.Data) > 0 {
+		return deck.Data[0].Id, nil
+	}
+
+	return "", nil
+}
+
+func retrieveLastDecks(set int, count int, offset int) (*vault.DecksVault, error) {
+	filter := ""
+	if set > 0 {
+		filter = fmt.Sprintf("&expansion=%d", set)
+	}
+	page := 1 + offset
+	url := fmt.Sprintf("https://www.keyforgegame.com/api/decks?page=%d&page_size=%d&ordering=-date%s", page, count, filter)
+	method := "GET"
+
+	client := &http.Client{}
```

Comment:
Use a custom HttpClient configuration to avoid hanging requests:
```
var transport = &http.Transport{
  Dial: (&net.Dialer{
    Timeout: 5 * time.Second,
  }).Dial,
  TLSHandshakeTimeout: 5 * time.Second,
}
var client = &http.Client{
  Timeout: time.Second * 10,
  Transport: transport,
}
```